src/txn/txn_ckpt.c

```diff
@@ -1485,6 +1488,32 @@ __checkpoint_lock_dirty_tree(
     return (ret);
 }
 
+/*
+ * __checkpoint_apply_obsolete --
+ *     Returns true if the checkpoint is obsolete.
+ */
+static bool
+__checkpoint_apply_obsolete(WT_SESSION_IMPL *session, WT_BTREE *btree, WT_CKPT *ckpt)
+{
+    if (ckpt->size != 0) {
+        /*
+         * Mark the btree has obsolete page if the checkpoint has valid stop timestamp. This flag is
```

Comment:
That first sentence does not make sense. How about:
"If the checkpoint has a valid stop timestamp, mark the btree as having obsolete pages."

Reply:
Fixed.